backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/predicate/impl/AbstractElasticsearchNestablePredicate.java

```diff
@@ -34,26 +34,26 @@ public void checkNestableWithin(String expectedParentNestedPath) {
 	}
 
 	@Override
-	public JsonObject toJsonQuery(PredicateRequestContext context) {
-		checkNestableWithin( context.getNestedPath() );
-
+	public JsonObject buildJsonQuery(PredicateRequestContext context) {
 		List<String> nestedPathHierarchy = getNestedPathHierarchy();
 		String expectedNestedPath = nestedPathHierarchy.isEmpty()
 				? null
 				: nestedPathHierarchy.get( nestedPathHierarchy.size() - 1 );
 
 		if ( Objects.equals( context.getNestedPath(), expectedNestedPath ) ) {
 			// Implicit nesting is not necessary
-			return super.toJsonQuery( context );
+			checkNestableWithin( context.getNestedPath() );
+			return super.buildJsonQuery( context );
 		}
 
 		// The context we expect this predicate to be built in.
 		// We'll make sure to wrap it in nested predicates as appropriate in the next few lines,
 		// so that the predicate is actually executed in this context.
 		PredicateRequestContext contextAfterImplicitNesting =
 				context.withNestedPath( expectedNestedPath );
+		checkNestableWithin( expectedNestedPath );
```

Comment:
btw, note this change: previously we'd just assume that the nested path would be ok for the predicate without double-checking.